app/models/user.rb

```diff
@@ -159,6 +159,10 @@ def authorised_applications
   end
   alias_method :applications_used, :authorised_applications
 
+  def revoke_all_authorisations
+    authorisations.each(&:revoke)
```

Comment:
Hi @mike29736 - I've just realised that this will update the revoked_at timestamp for _all_ tokens, including those that are already revoked. I think it should probably only update those that are not currently revoked so that we preserve the revoked_at timestamp of any tokens that have already been revoked. What do you think?

Reply:
Thanks, good catch